app/Services/ProductService.php

```diff
@@ -223,13 +226,44 @@ public function generateReportForProductStatesGraph($request)
         return parent::generateReport($parameters, $fileName, $request);
     }
 
-    public function uploadFile($request)
+
+    public function upload(Request $request)
     {
-        if ($request->has('mycsv')) {
-            $file = $request->file('mycsv');
-            $batchId = parent::upload($file, ProductCsvProcess::class);
-            return response()->json(['batch_id' => $batchId]);
+        Log::info($request);
+
+        $stream = fopen('php://input', 'r');
+
+        if (!$stream) {
+            return response()->json(['error' => 'Failed to open the stream'], 500);
+        }
+
+        try {
+            $filePath = storage_path('app/uploads/tmp.csv');
+
+            file_put_contents($filePath, $stream);
+
+            fclose($stream);
+
+            DB::beginTransaction();
+
+            DB::statement("
+            COPY products(name, description, product_type_id, created_at, updated_at, \"validFrom\", \"validTo\", status)
+            FROM '{$filePath}'
+            DELIMITER ','
+            CSV HEADER;
+        ");
+
+            DB::commit();
+
+            return response()->json(['message' => 'File upload processing completed']);
+        } catch (\Exception $e) {
+            DB::rollBack();
+
+            // Log the error
+            Log::error("Error processing file upload: " . $e->getMessage());
+
+            // Return a 500 error response
+            return response()->json(['error' => 'File upload processing failed'], 500);
```

Comment:
Could you please implement a logic for removing csv file?

Reply:
the logic has been implemented, please double check it again. cheers